web/src/message_view_header.ts

```diff
@@ -39,25 +41,39 @@ function get_message_view_header_context(filter: Filter | undefined): MessageVie
     if (recent_view_util.is_visible()) {
         return {
             title: $t({defaultMessage: "Recent conversations"}),
+            description: $t({defaultMessage: "Overview of ongoing conversations."}),
             zulip_icon: "recent",
+            link: "/help/recent-conversations",
         };
     }
     if (inbox_util.is_visible()) {
         return {
             title: $t({defaultMessage: "Inbox"}),
+            description: $t({
+                defaultMessage: "Overview of your conversations with unread messages.",
+            }),
             zulip_icon: "inbox",
+            link: "/help/inbox",
         };
     }
-    if (filter === undefined) {
+    if (filter === undefined || filter.is_in_home()) {
```

Comment:
`filter` is only `undefined` on an initial load, a hash change (`/#feed`) presents a filter object, so this OR statement checks for that condition, so that the Combined feed gets its proper description.